### DISCUSSION TITLE
Eigenvalues of non-Hermitian Hamiltonian

### DISCUSSION THREAD

**Discussion by wingcyu** [QuSpin/QuSpin#575] [Q&A]
Hello, may I know if Quspin can find the eigenvalues and eigenstates of a non-Hermitian Hamiltonian? From the user guide, it seems that the diagonalization methods (eigh, eigsh, eigvalsh) associated with the Hamiltonian only work for hermitian matrices. Is there a way to find the eigenvalues and eigenstates after constructing the Hamiltonian?

**Comment by JiahaoYao** [comment]
Hi @wingcyu , you can just use the quspin to generate the hamiltonian (matrix), and then use numpy to diagonalize the non-hermitian matrix, like eig or eigs.

**Reply by thefacemyer** [reply]
I'm trying the same thing, but the Hamiltonian object is something strange, not the usual array. How do we have QuSpin create the matrix and then turn it into an array so that we can execute LA.eig(H)?

**Reply by JiahaoYao** [reply]
How about H.toarray()

**Reply by thefacemyer** [reply]
Wow, I'm an idiot. I just emailed you, and I apologize. I was just thinking that we needed to maneuver around QuSpin objects using QuSpin objects. This seems to work.

**Reply by JiahaoYao** [reply]
No worries. Probably this should be written in the documentations @mgbukov. And feel free to let us know if there is further questions.
